docs/products/postgresql/concepts/pg-connection-pooling.rst

```diff
@@ -0,0 +1,92 @@
+        classDef style_client fill:lightgray,stroke:#333,stroke-width:0px;
+        pg_client_1(PG client) <-.->|Client establishing a new connection| pgbouncer
+        pg_client_2(PG client) ---> pgbouncer
+        pg_client_3(PG client) ---> pgbouncer
+        pg_client_4(PG client) ---> pgbouncer
+        pg_client_5(PG client) --->|Existing client connections| pgbouncer
+        pg_client_6(PG client) ---> pgbouncer
+        pg_client_7(PG client) ---> pgbouncer
+        pgbouncer --> postmaster
+        pgbouncer --> pg_backend_1(PG Backend 1)
+        pgbouncer --> pg_backend_2(PG Backend 2)
+        pgbouncer --> pg_backend_3(PG Backend 3)
+        pgbouncer --> pg_backend_4(PG Backend 4)
+
+
+    subgraph PostgreSQL server
+        pgbouncer
+        postmaster & pg_backend_1 & pg_backend_2 & pg_backend_3 & pg_backend_4
+    end
+
+    subgraph Clients
+        pg_client_1 & pg_client_2 & pg_client_3 & pg_client_4 & pg_client_5 & pg_client_6 & pg_client_7
+    end
+    class pg_client_1 style_main_node;
+    class pg_backend_1,pg_backend_2,pg_backend_3,pg_backend_4,pg_backend_5,pg_backend_6,postmaster,pgbouncer style_pg_services;
+    class Clients style_client;
+
+Many frameworks and libraries (ORMs, Django, Rails, etc.) support client-side pooling, which solves much the same problem. However, when there are many distributed applications or devices accessing the same database, a client-side solution is not enough.
+
+Connection pooling modes
+------------------------
+
+Aiven PostgreSQL supports three different operational pool modes: ``session``, ``transaction`` and ``statement``.
+
+* The ``session`` pooling mode means that once a client connection is granted access to a PostgreSQL server-side connection, it can hold it until the client disconnects from the pooler. After this, the server connection is added back onto the connection pooler's free connection list to wait for its next client connection. Client connections are accepted (at TCP level), but their queries only proceed once another client disconnects and frees up its backend connection back into the pool. This mode can be helpful in some cases for providing a wait queue for incoming connections while keeping the server memory usage low, but is of limited use under most common scenarios due to the slow recycling of the backend connections.
+* The ``transaction`` pooling mode allows each client connection to take their turn in using a backend connection for the duration of a single transaction. After the transaction is committed, the backend connection is returned back into the pool and the next waiting client connection gets to reuse the same connection immediately. In practice, this provides quick response times for queries as long as the typical execution times for transactions are not excessively long. This is the most commonly used PGBouncer mode and also the default pooling mode in Aiven for PostgreSQL.
```

Comment:
Let's put this first on the list and start with "the default and recommended option is ...." so that hasty readers pick an option that will probably work!